src/pages/Vote/VotingWallet/index.tsx

```diff
@@ -64,20 +133,63 @@ export const VoteUi: React.FC<IVoteUiProps> = ({ votingWeight, xvsLockedWei }) =
           <Trans
             i18nKey="vote.delegateYourVoting"
             components={{
-              Anchor: <span css={styles.clickableText} role="button" aria-pressed="false" />,
+              Anchor: (
+                <span
+                  css={styles.clickableText}
+                  role="button"
+                  aria-pressed="false"
+                  tabIndex={0}
+                  onClick={() => setDelegateModelIsOpen(true)}
+                />
+              ),
             }}
           />
         </Typography>
       </Paper>
+      <DelegateModal
+        onClose={() => setDelegateModelIsOpen(false)}
+        isOpen={delegateModelIsOpen}
+        currentUserAccountAddress={currentUserAccountAddress}
+        previouslyDelegated={previouslyDelegated}
+        setVoteDelegation={setVoteDelegation}
+      />
     </div>
   );
 };
 
-const Vote: React.FC = () => (
-  <VoteUi
-    votingWeight={new BigNumber(1.003)}
-    xvsLockedWei={new BigNumber('19931200345567000000')}
-  />
-);
+const VotingWallet: React.FC = () => {
+  const { account: { address: accountAddress } = { address: undefined }, openAuthModal } =
+    useContext(AuthContext);
+
+  const { data: currentVotesWei } = useGetCurrentVotes(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+  const { data: delegate } = useGetVoteDelegates(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+
+  const { data: vaults } = useGetVestingVaults({ accountAddress });
+  const [xvsVault] = vaults.filter(v => v.stakedTokenId === XVS_TOKEN_ID);
+
+  const { mutateAsync: setVoteDelegation } = useSetVoteDelegate({
```

Comment:
We'll want to track the loading state of the mutation (and display a loader on the submit button of the "Delegate/Redelegate voting" modal).
We'll also want to display a toast in case of an error, or a successful transaction modal in case of a success (on top of closing the "Delegate/Redelegate voting" modal). You can find the content for the successful transaction modal here: https://www.figma.com/file/GYJwWWLbC4B8DrJOPcETz3/Venus-2.0?node-id=5723%3A43139 (although I guess it should say "You successfully delegate**d** 8000 BNB)